SoObjects/SOGo/SOGoUser.m

```diff
@@ -342,18 +342,34 @@ - (NSString *) cn
 
 - (NSMutableDictionary *) defaultIdentity
 {
-  NSMutableDictionary *currentIdentity, *defaultIdentity;
-  NSEnumerator *identities;
+  NSDictionary *defaultAccount, *currentIdentity;
+  NSMutableDictionary *defaultIdentity;
+  NSArray *identities;
+  NSString *defaultEmail;
+  unsigned int count, max;
 
+  defaultEmail = [NSString stringWithFormat: @"%@@%@", [self loginInDomain], [self domain]];
+  defaultAccount = [[self mailAccounts] objectAtIndex: 0];
   defaultIdentity = nil;
 
-  identities = [[self allIdentities] objectEnumerator];
-  while (!defaultIdentity
-	 && (currentIdentity = [identities nextObject]))
-    if ([[currentIdentity objectForKey: @"isDefault"] boolValue])
-      defaultIdentity = currentIdentity;
+  identities = [defaultAccount objectForKey: @"identities"];
+  max = [identities count];
 
-  return defaultIdentity;
+  for (count = 0; count < max; count++)
+    {
+      currentIdentity = [identities objectAtIndex: count];
+      if ([[currentIdentity objectForKey: @"isDefault"] boolValue])
+        {
+          defaultIdentity = [NSMutableDictionary dictionaryWithDictionary: currentIdentity];
+          break;
+        }
+      else if ([[currentIdentity objectForKey: @"email"] caseInsensitiveCompare: defaultEmail] == NSOrderedSame)
+        {
+          defaultIdentity = [NSMutableDictionary dictionaryWithDictionary: currentIdentity];
```

Comment:
Should you also break here?

Reply:
No, because I can find an identity with the `isDefault` attribute later.